scalafmt-tests/src/test/resources/scala3/OptionalBraces.stat

```diff
@@ -1557,3 +1557,50 @@ val (length, width) =
      (hourMarkLength, hourMarkWidth)
   else
      (secondMarkLength, secondMarkWidth)
+<<< if indentation
+if // comm
+    val cond1 = true
+    (a1 && a2) == cond1
+  then
```

Comment:
here and in other places: can you add some tests with `danglingParentheses.ctrlSite = false`? when this setting is `true`, formatter forces a break before `then`; otherwise, it doesn't, and I would like to make sure scalameta can still parse it. otherwise, we might need to change the logic around `danglingParentheses.ctrlSite = false` when `if` condition was a block.

Reply:
This works with a block, but seems to break with a single statement if we have a comment:
```
danglingParentheses.ctrlSite = false
===
if // comm
    (a1 && a2) == cond1
  then
    val exitCode = 1
    exitCode
```

gets formatted to 
```scala
if // comm
   (a1 && a2) == cond1 then
```

but that is correct compiler wise, just doesn't work in the parser yet. I will add the tests.

Reply:
Added tests for `danglingParentheses`, let me know if anything is missing.

Reply:
so does the danglingParens setting affect formatting? the idea was that `false` should lead to no break before `then`, if compiler allows, of course.

Reply:
`danglingParens.ctrlSite` seem to only affect the formatting if the condition starts with `(` there is no special handling for optional braces.

We could do it later one after https://github.com/scalameta/scalameta/pull/2384 is released.

Reply:
What do you think @kitbellew ? Merge this and work on `danglingParens.ctrlSite` later or wait for the release and try to handle it all here?

Reply:
let's merge and then follow up.